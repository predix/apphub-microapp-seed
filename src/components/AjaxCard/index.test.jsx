import React from 'react';
import renderer from 'react-test-renderer';
// import nock from 'nock';
import axios from 'axios';
import { shallow } from 'enzyme';

import Card from '.';

jest.mock('axios');

describe('Ajax Card', () => {
  test('should render', () => {
    const wrapper = shallow(<Card headerText="My Header" />);
    expect(wrapper.find('.ajax-card')).toHaveLength(1);
  });
  test('should match snapshot', () => {
    const tree = renderer.create(<Card headerText="My Header">Content</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should update state on URL change', () => {
    const testUrl = 'http://localhost:80/api/test';
    // const baseUrl = 'http://localhost:80';
    const resp = { data: ['one', 'two', 'three'] };
    // const scope = nock(baseUrl).get('/api/test').reply(200, resp);
    // console.log(scope.pendingMocks());

    // Mock axios get
    axios.get.mockResolvedValue({ data: resp });

    const wrapper = shallow(<Card headerText="My Header" />);
    expect(wrapper.find('.ajax-card')).toHaveLength(1);

    // Input
    const inputElement = wrapper.find('#urlInput');
    expect(inputElement).toHaveLength(1);

    // Input Change
    inputElement.simulate('change', { target: { value: testUrl } });
    expect(wrapper.state('ajaxEndpoint')).toBe(testUrl);

    // Form submit
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(wrapper.state('pendingRequest')).toBe(true);

    // Form response
    console.log(wrapper.state());
    // expect(wrapper.state('ajaxData').data).toHaveLength(3);
  });
});
