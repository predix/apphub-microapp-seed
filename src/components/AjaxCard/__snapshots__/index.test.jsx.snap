// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ajax Card should match snapshot 1`] = `
<div
  className="px-card ajax-card"
>
  <div
    className="px-card sc-EHOje fJIaeZ"
  >
    <header
      className="px-card-header sc-bZQynM IGYNF"
    >
      <span
        className="sc-gzVnrw hNuuCV"
      >
        My Header
      </span>
    </header>
    <section
      className="px-card-body sc-htoDjs hSRIGh"
    >
      <h4>
        Example Routes
      </h4>
      <div
        className="sc-bwzfXH jJPvbV"
        wrap={true}
      >
        <div>
          <button
            className="px-button sc-bdVaJa ienWKz"
            icon={null}
            onClick={[Function]}
            size={null}
            type="button"
          >
            ./api/db
          </button>
        </div>
        <div>
          <button
            className="px-button sc-bdVaJa ienWKz"
            icon={null}
            onClick={[Function]}
            size={null}
            type="button"
          >
            ./api/db/_all_docs
          </button>
        </div>
        <div>
          <button
            className="px-button sc-bdVaJa ienWKz"
            icon={null}
            onClick={[Function]}
            size={null}
            type="button"
          >
            ./api/example
          </button>
        </div>
        <div>
          <button
            className="px-button sc-bdVaJa ienWKz"
            icon={null}
            onClick={[Function]}
            size={null}
            type="button"
          >
            ./api/example/getCookies
          </button>
        </div>
        <div>
          <button
            className="px-button sc-bdVaJa ienWKz"
            icon={null}
            onClick={[Function]}
            size={null}
            type="button"
          >
            ./api/example/setCookie/foo/bar
          </button>
        </div>
        <div>
          <button
            className="px-button sc-bdVaJa ienWKz"
            icon={null}
            onClick={[Function]}
            size={null}
            type="button"
          >
            ./api/example/timeout/1000
          </button>
        </div>
        <div>
          <button
            className="px-button sc-bdVaJa ienWKz"
            icon={null}
            onClick={[Function]}
            size={null}
            type="button"
          >
            ./api/example/500
          </button>
        </div>
        <div>
          <button
            className="px-button sc-bdVaJa ienWKz"
            icon={null}
            onClick={[Function]}
            size={null}
            type="button"
          >
            ./api/example/my/path/to/something
          </button>
        </div>
      </div>
      <br />
      <h4>
        Tester
      </h4>
      <form
        className="u-mb"
        onSubmit={[Function]}
      >
        <input
          className="sc-bxivhb hulgxF"
          id="urlInput"
          onChange={[Function]}
          placeholder="Enter URL..."
          type="text"
          value="./api/example?name=value"
        />
        <button
          className="px-button sc-bdVaJa ienWKz"
          icon={null}
          size={null}
          type="submit"
        >
          Send Request
        </button>
      </form>
    </section>
  </div>
</div>
`;
